lib/jqueue/exec.py

```diff
@@ -52,26 +52,28 @@
 from ganeti.jqueue import _JobProcessor
 
 
-def _GetMasterInfo():
-  """Retrieve job id, lock file name and secret params from the master process
+def _SetupJob():
+  """Setup the process to execute the job
 
-  This also closes standard input/output
+  Retrieve the job id from argv, create a livelock and pass it to the master
+  process and finally obtain any secret parameters.
+
+  This also closes standard input/output.
 
   @rtype: (int, string, json encoding of a list of dicts)
 
   """
+  job_id = int(sys.argv[1])
```

Comment:
I do wonder how things fail if this fails, or in general, if Python exits before communicating the livelock.

Reply:
If the Python process dies, then the pipe closes and the handshake will fail, raising an error on the master side. I though about guarding this piece of code a bit more, but there's not much to do if you don't get a valid job ID.

Reply:
Oh, I don't mean this needs to be hardened. I was just wondering if the error reporting on master side is good enough so that it can be debugged, that's all. And not really related to your change, to be honest.

Reply:
Before this changeset, nothing would be logged and you would have to dig pretty deep. I think I actually *had* to debug this for #1404 and it wasn't straightforward. After this PR, it will report stuff in LuxiD's stderr, which is redirected to the log AFAICT.

Reply:
Ack, SGTM!